test/integration/fake_upstream.h

```diff
@@ -315,6 +315,8 @@ class SharedConnectionWrapper : public Network::ConnectionCallbacks,
     if (disconnected_) {
       return testing::AssertionSuccess();
     }
+    ASSERT(!connection_.dispatcher().isThreadSafe(),
+           "deadlock: executeOnDispatcher called from dispatcher thread.");
```

Comment:
For my understanding, what changes in this PR that this is now required? It's not immediately clear. More comments?

Reply:
Added a comment.  Changes in this PR do not require this ASSERT.  I think I called this method from the dispatcher thread accidentally during development and had to debug why tests hit hard timeouts.